share/gpodder/extensions/subscription_stats.py

```diff
@@ -0,0 +1,185 @@
+        tree.append_column(channelcolumn)
+
+        lastcell = Gtk.CellRendererText()
+        lastcolumn = Gtk.TreeViewColumn(_('Last Updated'))
+        lastcolumn.set_sort_column_id(4)
+        lastcolumn.pack_end(lastcell, True)
+        lastcolumn.add_attribute(lastcell, 'text', 3)
+        lastcolumn.set_expand(True)
+        tree.append_column(lastcolumn)
+
+        notebook.append_page(scrolled,
+                             Gtk.Label('%d %s' % (len(channels), category)))
+
+    def open_dialog(self):
+        db = self.gpodder.db
+
+        # get all channels
+        channels = []
+        with db.lock:
+            cur = db.cursor()
+            cur.execute(
+                        'SELECT id, title, pause_subscription FROM %s'
+                        % db.TABLE_PODCAST)
+            while True:
+                row = cur.fetchone()
+                if row is None:
+                    break
+                channels.append(row)
+            cur.close()
+
+        # get average time between episodes per channel
+        now = int(time.time())
+        nr_paused = 0
+        daily = []
+        weekly = []
+        monthly = []
+        yearly = []
+        for channel_id, channel_name, paused in channels:
+            if paused:
+                nr_paused += 1
+
+            total = 0
+            nr_episodes = 0
+            prev = now
+            with db.lock:
+                cur = db.cursor()
+                cur.execute('SELECT published FROM %s WHERE podcast_id = %d \
+                            ORDER BY published DESC LIMIT 25'
+                            % (db.TABLE_EPISODE, channel_id))
+                while True:
+                    row = cur.fetchone()
+                    if row is None:
+                        break
+                    if total == 0:
+                        edate = row[0]
+                    total += (prev - row[0])
+                    nr_episodes += 1
+                    prev = row[0]
+                cur.close()
+
+            average = (total / nr_episodes) / (24 * 60 * 60) \
+                if nr_episodes > 0 else 0
+
+            if average == 0:
+                yearly.append([average, channel_name, edate, paused])
+            elif average <= 2:
+                daily.append([average, channel_name, edate, paused])
+            elif average <= 14:
+                weekly.append([average, channel_name, edate, paused])
+            elif average <= 61:
+                monthly.append([average, channel_name, edate, paused])
+            else:
+                yearly.append([average, channel_name, edate, paused])
+
+        # sort by averages
+        daily.sort(key=lambda e: e[0])
+        weekly.sort(key=lambda e: e[0])
+        monthly.sort(key=lambda e: e[0])
+        yearly.sort(key=lambda e: e[0])
+
+        # open dialog
+        dlg = Gtk.Dialog(_('Subscription Statistics'),
+                         self.gpodder.main_window)
+        dlg.set_size_request(400, 400)
```

Comment:
`set_size_request()` sets a fixed minimum size, so shrinking the dialog does not work. It's better to use `set_default_size()`. Setting `dlg.set_default_size(420, 600)` looked good at least on my screen with all the notebook tabs visible.

Reply:
The 400,400 was suggested from someone because of phones, but 420,600 maybe ok also for phones, so I changed this.